carp.common/src/commonTest/kotlin/dk/cachet/carp/common/application/NamespacedIdTest.kt

```diff
@@ -26,6 +26,43 @@ class NamespacedIdTest
         }
     }
 
+    @Test
+    fun name_may_not_contain_non_alphanumerics()
+    {
+        assertFailsWith<IllegalArgumentException>
+        {
+            NamespacedId( "some.namespace", "contains-hyphen!" )
+        }
+    }
+
+    @Test
+    fun namespace_cannot_end_with_dot()
+    {
+        assertFailsWith<IllegalArgumentException>
+        {
+            NamespacedId("some.namespace.", "name")
```

Comment:
Detekt might throw some formatting errors here (spaces should be needed). I can take care of those, don't worry. :)